Backpack-SwiftUI/ImageGalleryGrid/Classes/BPKImageGalleryGrid.swift

```diff
@@ -0,0 +1,230 @@
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+struct ImageGalleryGrid<ImageView: View>: ViewModifier {
+    let categories: [BPKImageGalleryCategory<ImageView>]
+    let closeAccessibilityLabel: String
+    let onCloseTapped: () -> Void
+    // @Binding var currentIndex: Int // Category Selected binding?
+    @Binding var isPresented: Bool
+    
+    func body(content: Content) -> some View {
+        content
+            .fullScreenCover(isPresented: $isPresented, content: {
+                ContentView(categories: categories)
+            })
+    }
+    
+    struct ContentView: View {
+        let categories: [BPKImageGalleryCategory<ImageView>]
+        @State var selectedCategory: Int = 0
+        var body: some View {
+            VStack(spacing: .md) {
+                ImageGalleryGridCategoriesCarousel(
+                    categories: categories,
+                    selectedCategory: $selectedCategory
+                )
+                .fixedSize(horizontal: false, vertical: true)
+                
+                ImageGalleryGridImagesGrid(
+                    images: categories[selectedCategory].images
+                )
+            }
+        }
+    }
+}
+
+public extension View {
+    func bpkImageGalleryGrid<Content>(
+        isPresented: Binding<Bool>,
+        categories: [BPKImageGalleryCategory<Content>],
+        closeAccessibilityLabel: String,
+        // currentIndex: Binding<Int>, // Category Selected binding?
+        onCloseTapped: @escaping () -> Void
+    ) -> some View {
+        modifier(
+            ImageGalleryGrid(
+                categories: categories,
+                closeAccessibilityLabel: closeAccessibilityLabel,
+                onCloseTapped: onCloseTapped,
+                isPresented: isPresented
+            )
+        )
+    }
+}
+
+struct ImageGalleryGridCategoriesCarousel<ImageView: View>: View {
+    let categories: [BPKImageGalleryCategory<ImageView>]
+    @Binding var selectedCategory: Int
+    private let categoryImageSideLength: CGFloat = 90
+    
+    var body: some View {
+        ScrollView(.horizontal) {
+            HStack(alignment: .top) {
+                ForEach(0..<categories.count, id: \.self) { ndx in
+                    categoryThumb(for: categories[ndx])
+                        .onTapGesture {
+                            selectedCategory = ndx
+                        }
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private func categoryThumb(for category: BPKImageGalleryCategory<ImageView>) -> some View {
+        VStack {
+            category.categoryImage.content()
+                .frame(
+                    width: categoryImageSideLength,
+                    height: categoryImageSideLength
+                )
+                .clipShape(RoundedRectangle(cornerRadius: .md))
+            BPKText(category.title)
+                .lineLimit(nil)
+        }
+        .frame(width: categoryImageSideLength)
```

Comment:
would it work just setting width and height here and not in the category image?

Reply:
Unfortunately not, at least from my testing anyway we need this here, and also Bothe width and height frame set on the image